prow/plugins/wip/wip-label_test.go

```diff
@@ -0,0 +1,219 @@
+			hasWipLabelComment: true,
+			shouldLabel:        false,
+			shouldUnlabel:      false,
+			shouldComment:      false,
+			shouldUncomment:    false,
+		},
+	}
+	for _, tc := range testcases {
+		fc := &fakegithub.FakeClient{
+			PullRequests:  make(map[int]*github.PullRequest),
+			IssueComments: make(map[int][]github.IssueComment),
+		}
+		org, repo, number, id, body := "org", "repo", 5, 0, "body"
+		e := &event{
+			org:                org,
+			repo:               repo,
+			number:             number,
+			hasWipLabel:        tc.hasWipLabel,
+			needsWipLabel:      tc.needsWipLabel,
+			hasWipLabelComment: tc.hasWipLabelComment,
+			wipLabelCommentID:  id,
+			wipLabelBody:       body,
+		}
+		if tc.shouldUncomment {
+			fc.IssueComments[number] = []github.IssueComment{{ID: id}}
+		}
+
+		if err := handle(fc, logrus.WithField("plugin", pluginName), e); err != nil {
+			t.Errorf("For case %s, didn't expect error from wip: %v", tc.name, err)
+			continue
+		}
+
+		fakeLabel := fmt.Sprintf("%s/%s#%d:%s", org, repo, number, wipLabel)
+		if tc.shouldLabel {
+			if len(fc.LabelsAdded) != 1 || fc.LabelsAdded[0] != fakeLabel {
+				t.Errorf("For case %s: expected to add %q label but instead added: %v", tc.name, wipLabel, fc.LabelsAdded)
+			}
+		} else if len(fc.LabelsAdded) > 0 {
+			t.Errorf("For case %s, expected to not add %q label but added: %v", tc.name, wipLabel, fc.LabelsAdded)
+		}
+		if tc.shouldUnlabel {
+			if len(fc.LabelsRemoved) != 1 || fc.LabelsRemoved[0] != fakeLabel {
+				t.Errorf("For case %s: expected to remove %q label but instead removed: %v", tc.name, wipLabel, fc.LabelsRemoved)
+			}
+		} else if len(fc.LabelsRemoved) > 0 {
+			t.Errorf("For case %s, expected to not remove %q label but removed: %v", tc.name, wipLabel, fc.LabelsRemoved)
+		}
+
+		if tc.shouldComment {
+			if len(fc.IssueCommentsAdded) != 1 || fc.IssueCommentsAdded[0] != fmt.Sprintf("%s/%s#%d:%s", org, repo, number, body) {
+				t.Errorf("For case %s: expected to add comment but instead added: %v", tc.name, fc.IssueCommentsAdded)
+			}
+		} else if len(fc.IssueCommentsAdded) > 0 {
+			t.Errorf("For case %s, expected to not add comment but added: %v", tc.name, fc.LabelsAdded)
+		}
+		if tc.shouldUncomment {
+			if len(fc.IssueCommentsDeleted) != 1 || fc.IssueCommentsDeleted[0] != fmt.Sprintf("%s/%s#%d", org, repo, id) {
+				t.Errorf("For case %s: expected to remove comment but instead removed: %v", tc.name, fc.IssueCommentsDeleted)
+			}
+		} else if len(fc.IssueCommentsDeleted) > 0 {
+			t.Errorf("For case %s, expected to not remove comment but removed: %v", tc.name, fc.IssueCommentsDeleted)
+		}
+	}
+}
+
+func TestHasWipPrefix(t *testing.T) {
+	var tests = []struct {
+		title    string
+		expected bool
+	}{
+		{
+			title:    "dummy title",
+			expected: false,
+		},
+		{
+			title:    "WIP dummy title",
+			expected: true,
+		},
+		{
+			title:    "[WIP] dummy title",
+			expected: true,
+		},
+		{
+			title:    "<WIP> dummy title",
```

Comment:
I couldn't find any instances of <WIP>... can we drop it? What do you think of picking either `WIP` or `[WIP]` and requiring it to match one of these instead of either?

Reply:
I don't much care -- I'd rather make it easier rather than harder. We could just look for `WIP` somewhere in the title and disregard it's position and/or enclosing brackets.

Reply:
Personally I think it is hard to write flexible software. It is much easier to write good, simple software. And it is much easier for humans to adapt to the expectations of the software than visa versa. So I would aim to make these commands more like `ls` and `git` than the Google Assistant and/or Alexa.

And as amazing as these assistants are their limitations and understanding human intent become immediately obvious as soon you use them. I pretty clearly adapt how I talk to the Google Assistant as I learn what it can do and what it likes.